vispy/visuals/_scalable_textures.py

```diff
@@ -295,29 +295,25 @@ def clim_normalized(self):
 
     @staticmethod
     def _scale_data_on_cpu(data, clim, copy=True):
-        if copy:
-            should_cast_to_f32(data.dtype)
-            data = np.array(data, dtype=np.float32, copy=copy)
-        elif not copy and not np.issubdtype(data.dtype, np.floating):
-            raise ValueError("Data must be of floating type for no copying to occur.")
-
+        data = np.array(data, dtype=np.float32, copy=copy)
         if clim[0] != clim[1]:
             data -= clim[0]
-            data *= 1.0 / (clim[1] - clim[0])
-        if should_cast_to_f32(data.dtype):
-            data = data.astype(np.float32)
```

Comment:
Note that the returned data *must* be f32 here. This means that in the previous logic, no matter what, if the data was *not* f32, it was being converted.

So this is exactly the same now, except we let `numpy` do the thinking; `np.array` already only copies if necessary. So if `copy` is `False`, and `dtype` is already `f32`, it won't copy.

```
copy : bool, optional
    If true (default), then the object is copied.  Otherwise, a copy will
    only be made if __array__ returns a copy, if obj is a nested sequence,
    or if a copy is needed to satisfy any of the other requirements
    (`dtype`, `order`, etc.).
```

Reply:
Ok. I trust you :wink: